src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -315,19 +317,65 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		});
 	}
 
-	private _getIcon(): Codicon | undefined {
-		if (this.shellLaunchConfig.icon) {
-			return iconRegistry.get(this.shellLaunchConfig.icon);
-		}
-		if (this.shellLaunchConfig?.attachPersistentProcess?.icon) {
-			return iconRegistry.get(this.shellLaunchConfig.attachPersistentProcess.icon);
+	// TODO:@meganrogge move all of this to profile resolver service
+	getIconUris(iconPath: string | { dark: URI, light: URI } | ThemeIcon): { dark: URI, light: URI } {
+		const dark = this.getDarkIconUri(iconPath as URI | { light: URI; dark: URI; });
+		const light = this.getLightIconUri(iconPath as URI | { light: URI; dark: URI; });
+		return {
+			dark: typeof dark === 'string' ? URI.file(dark) : dark,
+			light: typeof light === 'string' ? URI.file(light) : light
+		};
+	}
+
+	getLightIconUri(iconPath: URI | { light: URI; dark: URI; }) {
+		return typeof iconPath === 'object' && 'light' in iconPath ? iconPath.light : iconPath;
+	}
+
+	getDarkIconUri(iconPath: URI | { light: URI; dark: URI; }) {
+		return typeof iconPath === 'object' && 'dark' in iconPath ? iconPath.dark : iconPath;
+	}
+
+
+	private _getIcon(): Codicon | ThemeIcon | undefined | string {
```

Comment:
is this what we want it to return ?

string in the case of a custom (URI defined) icon so that it can be added to the extraClasses array in terminalTabsList